### DISCUSSION TITLE
matches(string, string) -> bool not working.

### DISCUSSION THREAD

**Discussion by leewjae** [google/cel-spec#247] [Q&A] [answered]
Hi folks, I wanted to use the function above to math regex, but it seems not working for me.
Fortunately, regex.matches(string) works. But I want to know how can I use the function above. Any ideas?
Thanks

**Comment by TristonianJones** [comment, marked as answer]
Hi @leewjae,

The official spec behavior for the member syntax is `string.matches(regex)`. The global function is supported in the cel-go runtime, but was omitted from the standard environment declarations, so it only worked if you didn't type-check the expression. I've put together a fix here: https://github.com/google/cel-go/pull/589

Thanks for bringing this to our attention.

-Tristan

**Reply by WMcKibbin** [reply]
https://github.com/google/cel-spec/blob/master/doc/intro.md

The documentation here is referencing the match(string,string) function

> CEL also supports construction of list, map, and protocol buffer objects. The following expressions evaluate to true:
>
>has(account.user_id) || has(account.gaia_id)    // true if either one is set
> size(account.emails) > 0                        // true if emails is non-empty
> matches(account.phone_number, "[0-9-]+")        // true if number matches regexp
